### PR TITLE
release/11.2.2

### PR DESCRIPTION
**Pull request by southbite**

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -1607,41 +1605,61 @@
...
-        return this.__performSystemRequest('disconnect-child-sessions', null, null, function(e) {
-          if (e)
-            _this.log.warn(
-              'socket.end failed in client, disconnect-child-sessions failed: ' + e.toString()
-            );
-          _this.__endSocket();
-        });
+    if (options.revokeToken || options.revokeSession)
+      return this.revokeToken(e => {
+        if (e) this.log.warn('socket.end failed in client, revoke-token failed: ' + e.toString());
```
Comment (JensEggers) [inline lib/client.js]: is socket.end the correct context message here?

File: lib/client.js [hunk 2]
```diff
@@ -1607,41 +1605,61 @@
...
 
-      this.__endSocket();
-    } catch (e) {
-      this.log.warn('socket.end failed in client: ' + e.toString());
-    }
+    if (options.disconnectChildSessions)
+      return this.__performSystemRequest('disconnect-child-sessions', null, null, e => {
+        if (e)
+          this.log.warn(
+            'socket.end failed in client, disconnect-child-sessions failed: ' + e.toString()
```
Comment (JensEggers) [inline lib/client.js]: is socket.end the correct context message here?

File: lib/client.js (hunk 1 above)
Comment (southbite) [reply lib/client.js]: sorted.

File: lib/client.js (hunk 2 above)
Comment (southbite) [reply lib/client.js]: fixed

**Pull request closed without merging**